ws/src/frida_language_processing/scripts/item_categorization.py

```diff
@@ -1,152 +1,98 @@
 #!/usr/bin/env python3
 """
-ROS node for the Storing groceries task, should return the category of an item or list of items
+ROS2 node that returns the best match of an item or list of items
 """
-import rospy
+import rclpy
 import pandas as pd
 import numpy as np
 import os
 import pickle
 from sentence_transformers import SentenceTransformer, util
 import yaml
 import tqdm
-import rospkg
-from openai import OpenAI
-
-from frida_hri_interfaces.srv import ItemsCategory, ItemsCategoryResponse
-
+import time
+import pynvml
+from frida_hri_interfaces.srv import ItemsCategory
+from ament_index_python.packages import get_package_share_directory
+import pynvml
+DATAFRAMES_DIR = os.path.join(get_package_share_directory('frida_language_processing'), 'scripts/dataframes/')
 # Minimum similarity percentage in which 2 words are considered the same
 SIMILARITY_THRESHOLD = 0.7
 
-rp = rospkg.RosPack()
-DATAFRAMES_DIR = rp.get_path(
-    'frida_language_processing') + "/scripts/dataframes/"
 ITEMS_EXTRACT_TOPIC = "items_category"
 OBJECT_EXTRACT_TOPIC = "object_category"
 
-ITEM_EXTRACT_TOPIC = "known_item"
 
 
 class ItemCategorization:
     """class to handle both services for item categorization"""
 
     def __init__(self) -> None:
         """Initialize the ROS node"""
-        self._node = rospy.init_node("item_categorization")
-        self._rate = rospy.Rate(10)
-        # rospy.Service(ITEMS_EXTRACT_TOPIC, ItemsCategory, self.extract_category)
-        rospy.Service(ITEMS_EXTRACT_TOPIC, ItemsCategory,
-                      self.get_local_embedding)
-        rospy.Service(OBJECT_EXTRACT_TOPIC, ItemsCategory, self.extract_item)
-        self.openai_client = OpenAI(
-            api_key=os.getenv("OPENAI_API_KEY")
-        )
+        self._node = rclpy.create_node("item_categorization")
+        #services
+        #self._node.create_service(ItemsCategory,ITEMS_EXTRACT_TOPIC, self.get_local_embedding)
+        self._node.create_service(ItemsCategory, OBJECT_EXTRACT_TOPIC , self.extract_item)
+        #load the model
         self.model = SentenceTransformer("all-MiniLM-L12-v2")
         current_dir = os.path.dirname(os.path.abspath(__file__))
+        #Load the embeddings already calculated/ embeddings database
         with open(os.path.join(current_dir, "assets", "embeddings_items.pkl"), "rb") as f:
             self.items_dataframe = pickle.load(f)
 
         with open(os.path.join(current_dir, "assets", "embeddings_categories.pkl"), "rb") as f:
             self.category_embeddings = pickle.load(f)
-
-        rospy.spin()
+        
+        with open(os.path.join(current_dir, "assets", "embeddings_locations.pkl"), "rb") as f:
+            self.locations_embeddings = pickle.load(f)
```

Comment:
self.category_embeddings, self.locations_embeddings, and self.actions_embeddings seem to be unused throughout the code and I think their corresponding files are missing / not commited (embeddings_categories.pkl, embeddings_locations.pkl, embeddings_actions.pkl)

Reply:
pushed changes